apis/nucleus/src/components/listbox/hooks/useDataStore.js

```diff
@@ -1,12 +1,18 @@
+import { useCallback } from 'react';
 import { useVizDataStore } from '../../../stores/viz-store';
 
 export default function useDataStore(model) {
   const keyPrefix = `${model.id}`;
 
   const [vizDataStore] = useVizDataStore();
 
-  const setStoreValue = (key, val) => vizDataStore.set(`${keyPrefix}/${key}`, val);
-  const getStoreValue = (key) => vizDataStore.get(`${keyPrefix}/${key}`);
+  const setStoreValue = useCallback(
+    (key, val) => {
+      vizDataStore.set(`${keyPrefix}/${key}`, val);
+    },
+    [vizDataStore]
```

Comment:
also include keyPrefix as dependency here and for getStoreValue
```suggestion
    [vizDataStore, keyPrefix]
```
(just to be extra safe, it will probably never change and therefore also work without)